internal/gitinterface/log_test.go

```diff
@@ -102,4 +98,182 @@ func TestGetCommitsBetweenRange(t *testing.T) {
+	if err != nil {
+		t.Fatal(err)
+	}
+
+	t.Run("Get all commits from merge commit 5", func(t *testing.T) {
+		commits, err := GetCommitsBetweenRange(repo, commitIDs[4], plumbing.ZeroHash)
+		expectedCommits := []*object.Commit{allCommits[4], allCommits[1], allCommits[0]}
+		sort.Slice(expectedCommits, func(i, j int) bool {
+			return expectedCommits[i].ID().String() < expectedCommits[j].ID().String()
+		})
+
+		assert.Equal(t, expectedCommits, commits)
+		assert.Nil(t, err)
+	})
+
+	t.Run("Get all commits from leaf commit 6 to root commit 1", func(t *testing.T) {
+		commits, err := GetCommitsBetweenRange(repo, commitIDs[5], plumbing.ZeroHash)
+		assert.Nil(t, err)
+		expectedCommits := []*object.Commit{allCommits[5], allCommits[2], allCommits[0]}
+		sort.Slice(expectedCommits, func(i, j int) bool {
+			return expectedCommits[i].ID().String() < expectedCommits[j].ID().String()
+		})
+		assert.Equal(t, expectedCommits, commits)
+	})
+}
+
+func GetCommitsFromCommitIds(commitIDs []plumbing.Hash, repo *git.Repository) ([]*object.Commit, error) {
+	allCommits := make([]*object.Commit, 0, len(commitIDs))
+	for _, commitID := range commitIDs {
+		commit, err := GetCommit(repo, commitID)
+		if err != nil {
+			return nil, err
+		}
+
+		allCommits = append(allCommits, commit)
+	}
+	return allCommits, nil
+}
+
+func applyCommitAndRefreshRef(t *testing.T, repo *git.Repository, commit *object.Commit, ref *plumbing.Reference) {
+	t.Helper()
+	if _, err := ApplyCommit(repo, commit, ref); err != nil {
+		t.Fatal(err)
+	}
+
+	newRef, err := repo.Reference(ref.Name(), true)
+	if err != nil {
+		t.Fatal(err)
+	}
+
+	*ref = *newRef
+}
+
+func createTreeHashes(repo *git.Repository, emptyBlobHash plumbing.Hash, num int) ([]plumbing.Hash, error) {
+	treeHashes := make([]plumbing.Hash, 0, num)
+	for i := 1; i <= num; i++ {
+		objects := make([]object.TreeEntry, 0, i)
+		for j := 0; j < i; j++ {
+			objects = append(objects, object.TreeEntry{Name: fmt.Sprintf("%d", j+1), Hash: emptyBlobHash})
+		}
+
+		treeHash, err := WriteTree(repo, objects)
+		if err != nil {
+			return nil, err
+		}
+
+		treeHashes = append(treeHashes, treeHash)
+	}
+	return treeHashes, nil
+}
+
+func createChildrenCommits(t *testing.T, repo *git.Repository, ref *plumbing.Reference, treeHashes []plumbing.Hash, numChildren int) ([]plumbing.Hash, []plumbing.Hash) {
+	t.Helper()
+
+	children := make([]plumbing.Hash, 0, numChildren)
+	commitIDs := make([]plumbing.Hash, 0, numChildren)
+
+	for i := 1; i <= numChildren; i++ {
+		commit := CreateCommitObject(testGitConfig, treeHashes[i], []plumbing.Hash{ref.Hash()}, fmt.Sprintf("Test commit %v", i+1), testClock)
+		childRefName := plumbing.ReferenceName(fmt.Sprintf("refs/heads/child%d", i))
+		childRef := plumbing.NewHashReference(childRefName, commit.Hash)
+
+		applyCommitAndRefreshRef(t, repo, commit, childRef)
+
+		commitIDs = append(commitIDs, childRef.Hash())
+		children = append(children, childRef.Hash())
```

Comment:
Are you returning this twice to separate handling children at during invocation? I wonder if some things could be moved around there instead?

Reply:
I had refactored the creation of children into the function, and I overlooked that, the result is now just one slice of hashes that is be appended to the commitID's slice in the main test.